### PR TITLE
[StructuralMechanicsApplication] Improving tangent pertubator utility

### PR DESCRIPTION
**Pull request by loumalouomega**
Correcting some errors, and adding second order perturbation

Adding new tests

### CODE REVIEW

**Comment by loumalouomega** [comment]
I added the option to choose via Properties which order do you want to consider and if consider the perturbation threshold

File: applications/StructuralMechanicsApplication/tests/cpp_tests/test_tangent_tensor_numerical_derivation_utility.cpp
Comment (frastellini) [inline applications/StructuralMechanicsApplication/tests/cpp_tests/test_tangent_tensor_numerical_derivation_utility.cpp:348]: I think we should also include a test for the first order perturbation.
Note: The use of second order perturbation is much expensive and not necessary better. Do we have any test about convergence speed between first and second order?

Comment (loumalouomega) [reply applications/StructuralMechanicsApplication/tests/cpp_tests/test_tangent_tensor_numerical_derivation_utility.cpp:348]: Indeed, the reason I did the second order is because the convergence rate was between 1.2-1.4, the second order is 1.97. That's why I did it

Comment (loumalouomega) [reply applications/StructuralMechanicsApplication/tests/cpp_tests/test_tangent_tensor_numerical_derivation_utility.cpp:348]: I can add the test

**Comment by loumalouomega** [comment]
Uuum, there is test failing, I check it before push it

**Review by philbucher — DISMISSED** [review]
blocking until #3567 is merged, otherwise the tests in StructuralMechanics are not run
Sorry for the inconvenience :/
I will merge master once #3567 is in

File: applications/StructuralMechanicsApplication/tests/cpp_tests/test_tangent_tensor_numerical_derivation_utility.cpp
Comment (frastellini) [reply applications/StructuralMechanicsApplication/tests/cpp_tests/test_tangent_tensor_numerical_derivation_utility.cpp:348]: Fantastic! Thank you!

**Comment by loumalouomega** [comment]
@AlejandroCornejo can you run the cpp tests in your machine?, in my machine work and here apparently doesn't

**Comment by loumalouomega** [comment]
@AlejandroCornejo The compilation issue is solved

**Review by AlejandroCornejo — APPROVED** [review]

**Pull request merged**
